Assignments/A1/Distro/AI_search.c

```diff
@@ -45,21 +45,136 @@
 
 #include "AI_search.h"
 
-// BEGIN STRUCT HELPER FUNCTION DEFINITIONS
-Deque Deque_new(void) {
-  Deque *deque = malloc(sizeof(Deque));
-  if (deque == NULL) {
-    fprintf(stderr, "malloc failed!\n");
-    exit(EXIT_FAILURE);
-  }
+// BEGIN STRUCT HELPER FUNCTION DEFS
+DequeItem *DequeItem_new(int x, int y) {
```

Comment:
Perhaps we can name this `create_DequeItem`? Same with `create_Deque` on line 59?